apps/mobile/src/libs/components/categories-form/categories-form.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback } from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import { useAppForm, useFormController } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	categoriesValidationSchema,
+	type CategoryDto,
+} from "~/packages/categories/categories";
+
+import { CATEGORIES_FORM_DEFAULT_VALUE } from "./libs/constants/constants";
+import { styles } from "./styles";
+
+type Properties = {
+	categories: CategoryDto[];
+	onSubmit: (selectedCategoryIds: number[]) => void;
+};
+
+const CategoriesForm: React.FC<Properties> = ({ categories, onSubmit }) => {
+	const { control, errors, handleSubmit } = useAppForm({
+		defaultValues: CATEGORIES_FORM_DEFAULT_VALUE,
+		validationSchema: categoriesValidationSchema,
+	});
+
+	const { field } = useFormController({
+		control,
+		name: "categories",
+	});
+
+	const { onChange, value: selectedCategories } = field;
+
+	const handleSaveCategories = useCallback((): void => {
+		onSubmit(selectedCategories);
+	}, [selectedCategories, onSubmit]);
+
+	const handleCheckboxChange = (categoryId: number) => (): void => {
+		const updatedCategories = selectedCategories.includes(categoryId)
+			? selectedCategories.filter((id) => id !== categoryId)
+			: [...selectedCategories, categoryId];
+		onChange(updatedCategories);
+	};
+
+	const handleAllChange = useCallback(
+		(isChecked: boolean): void => {
+			const updatedCategories = isChecked
+				? categories.map((category) => category.id)
+				: [];
+			onChange(updatedCategories);
+		},
+		[categories, onChange],
+	);
+
+	const handleFormSubmit = useCallback((): void => {
+		void handleSubmit(handleSaveCategories)();
```

Comment:
```suggestion
		void handleSubmit(onSubmit)();
```
The selected categories will be arguments to onSubmit